avm/res/compute/virtual-machine/main.bicep

```diff
@@ -98,14 +98,14 @@ param proximityPlacementGroupResourceId string = ''
 @description('Optional. Resource ID of an availability set. Cannot be used in combination with availability zone nor scale set.')
 param availabilitySetResourceId string = ''
 
-@description('Optional. If set to 1, 2 or 3, the availability zone for all VMs is hardcoded to that value. If zero, then availability zones is not used. Cannot be used in combination with availability set nor scale set.')
+@description('Required. If set to 1, 2 or 3, the availability zone for all VMs is hardcoded to that value. If zero, then availability zones is not used. Cannot be used in combination with availability set nor scale set.')
 @allowed([
-  0
-  1
-  2
-  3
+  '0'
```

Comment:
I take it numbers are not supported? I wonder if we should keep the input across all modules as an integer and just convert the value to a string. Something to discuss in the team, I guess

Reply:
Chris mentioned, that the array in the Bicep reference for availability zones, is a string array, so I changed it to that

Reply:
I'll bring it up in the team :) to test the waters

Reply:
Alright, as discussed please change this back to the original int.

Reply:
Done